sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDBFileManager.scala

```diff
@@ -204,6 +206,14 @@ class RocksDBFileManager(
       case _ =>
         throw QueryExecutionErrors.invalidChangeLogReaderVersion(changelogVersion)
     }
+
+    // If the changelogFile is written under checkpoint v2 (checkpointUniqueId.isDefined)
+    // the first line would be the lineage. We should update the file position by
+    // reading from the lineage during the reader initialization.
+    if (checkpointUniqueId.isDefined) {
+      changelogReader.lineage
```

Comment:
I guess here we reference a variable to force it to be initialized? I think it would be easier to understand if we explicitly create a function such as `loadLineage()` or something like that.